### PR TITLE
Controller mapping detection

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
# XRTK - Mixed Reality Toolkit Change Request

## Overview

First part of resolving #431, to detect and check if a platform has any profiles configured.

Will still required each platform to have an editor script to check configuration on load, which I'm testing.
Next step is to also generate and add configuration automatically.

## Target of the change:

Is this enhancement for:

- Core (core framework, interfaces and definitions)

### CODE REVIEW

File: XRTK-Core/Packages/com.xrtk.core/Utilities/Editor/PathFinder.cs [hunk 1]
```diff
@@ -1,4 +1,7 @@
-﻿using UnityEngine;
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Utilities/Editor/PathFinder.cs]: This is ours not Microsofts.

File: XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs]: Is this copied from the MRTK?

File: XRTK-Core/Packages/com.xrtk.core/Utilities/Editor/PathFinder.cs (hunk 1 above)
Comment (SimonDarksideJ) [reply XRTK-Core/Packages/com.xrtk.core/Utilities/Editor/PathFinder.cs]: Ahh right, well I was just not removing it, curious as I remember you adding it.

File: XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs (hunk 2 above)
Comment (SimonDarksideJ) [reply XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs]: Bugger, good catch. No

File: XRTK-Core/Packages/com.xrtk.core/Utilities/Editor/PathFinder.cs (hunk 1 above)
Comment (SimonDarksideJ) [reply XRTK-Core/Packages/com.xrtk.core/Utilities/Editor/PathFinder.cs]: I read that wrong, Fixed

**Comment by StephenHodgson** [comment]
Your SDK submodule is pointing to a branch that might have been deleted. Try updating it to the latest development branch.

File: XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Text;
+using UnityEditor;
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs]: editor?

File: XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+using UnityEngine;
+using XRTK.Definitions.Utilities;
+using XRTK.Extensions;
+using XRTK.Providers.Controllers;
+using XRTK.Services;
+
+namespace XRTK.Utilities.Editor
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs]: Is this in a valid editor/inspector folder with an assembly definition?

Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs]: Namespace doesn't match the file location

File: XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs (hunk 3 above)
Comment (SimonDarksideJ) [reply XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs]: Removed editor namespace

File: XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs (hunk 4 above)
Comment (SimonDarksideJ) [reply XRTK-Core/Packages/com.xrtk.core/Utilities/ValidateConfiguration.cs]: removed editor namespace

**Comment by SimonDarksideJ** [comment]
After some serious review, I simply removed the editor namespace.  Moving it to the editor fold caused a whole heap of issues and cross dependency issues.
There a nested dependency whereby the main XRTK project is dependant on the Editor project, which in my view is wrong, but that is larger than the scope of this PR.

**Pull request merged**